From ddbf4adc2c500faf60a1ab81ad97aa8be081c1fe Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sun, 19 Dec 2021 20:57:39 +0000
Subject: [PATCH 4/4] gnu: Add Golang module - fasthttp

* gnu/packages/golang.scm: (go-github-com-valyala-fasthttp): New variable.
---
 gnu/packages/golang.scm | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 505feb997d..9f094cc08e 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -8916,3 +8916,34 @@ (define-public go-github-com-valyala-tcplisten
       "Package tcplisten provides customizable TCP net.Listener with various
 performance-related options:")
     (license license:expat)))
+
+(define-public go-github-com-valyala-fasthttp
+  (package
+    (name "go-github-com-valyala-fasthttp")
+    (version "1.31.0")
+    (source
+      (origin
+        (method git-fetch)
+        (uri (git-reference
+               (url "https://github.com/valyala/fasthttp")
+               (commit (string-append "v" version))))
+        (file-name (git-file-name name version))
+        (sha256
+          (base32 "0ra0n2shcp11736xv37cjnsqn32gvqfm3dkf9v8j98xmch2wqxqg"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "github.com/valyala/fasthttp"))
+    (propagated-inputs
+      (list go-golang-org-x-sys
+            go-golang-org-x-net
+            go-golang-org-x-crypto
+            go-github-com-valyala-tcplisten
+            go-github-com-valyala-bytebufferpool
+            go-github-com-klauspost-compress
+            go-github-com-andybalholm-brotli))
+    (home-page "https://github.com/valyala/fasthttp")
+    (synopsis "Provides fast HTTP server and client API")
+    (description
+     "This package provides a Go module @code{fasthttp} which may be used as
+replacement for native @code{net/http} module.")
+    (license license:expat)))
-- 
2.34.0

